### PR TITLE
Google search example

### PR DESCRIPTION
**Pull request by JoelEinbinder**
#178

### CODE REVIEW

**Comment by aslushnikov** [comment]
This example is a sub-example of pavel's stress-test, let's have that one instead when the #206 is fixed?

**Comment by JoelEinbinder** [comment]
I find the output of this easier to follow compared to pavel's stress test.

File: examples/search.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+(async() => {
+
+const {Browser} = require('..');
+const browser = new Browser();
+
+const page = await browser.newPage();
+await page.navigate('https://google.com', {waitUntil: 'networkidle'});
+// Type our query into the search bar
+await page.type('puppeteer');
+await page.press('Enter');
```
Comment (aslushnikov) [inline examples/search.js]: this doesn't work for me reliably - sometimes enter is "swallowed" and i'm stuck with the text in the text box.

File: examples/search.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+// Wait for the results to show up
+await page.waitForSelector('h3 a');
```
Comment (ebidel) [inline examples/search.js]: FWIW, other test runner examples I've seen look at the title to determine the results page is done.  Could also do that to be familiar: `await page.waitForFunction("document.title == 'puppeteer - Google Search'");`

Comment (aslushnikov) [reply examples/search.js]: The title changes at indeterminate point of time - whereas the `page.waitForSelector('h3 a')` waits for the first search result appearing on the page. Sounds more reliable!

**Comment by JoelEinbinder** [comment]
Sometimes if I run in headful mode, it only gets part of the search results because they stream in. Fixing that would make the example a bit more complicated, so I think the current state of it is fine.

File: examples/search.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+(async() => {
+
+const {Browser} = require('..');
```
Comment (aslushnikov) [inline examples/search.js]:
```js
require('puppeteer')
```

**Review by aslushnikov — APPROVED** [review]
Interestingly, works smoothly for me now

**Pull request merged**